lib/logstash/inputs/udp.rb

```diff
@@ -114,4 +128,21 @@ def stop
     @udp.close rescue nil
   end
 
+  def identity(client)
+    [client[3], client[1]].compact.join("-")
+  end
+
+  def need_identity_map?
+    case @codec
+    when LogStash::Codecs::Netflow
+      true
+    when LogStash::Codecs::IdentityMapCodec
+      true
+    when LogStash::Codecs::Multiline
+      true
+    else
+      false
+    end
+  end
+
```

Comment:
@ph - I think it was because the IdentityMapCodec was not a independent plugin (tied to multiline) and a bit experimental so I wanted to be able to make quick changes to it without needing to change core classes etc.
Further you and I thought (iirc) that if IMC proved a viable and settled solution and was applicable to more than the file and beats input we would refactor by dragging IMC into core as independent of ML codec.

But that was before Event Milling and the idea of codec split to decoder and encoder - because Event Milling will obsolete IMC as a codec wrapper. For this reason I advised @jorritfolmer to make a netflow input based on TCP/UDP input. @jordansissel did not know about my reasoning and advised Jorrit to add the IMC code to this codec.

Based on my limited understanding of the netflow codec operation it seems to me that an input would be able to provide a more flexible base on which to build the stateful template caching and lookup.